File: go/pkg/models/fete_room_team.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type FeteRoomTeam struct {
	ID         uuid.UUID      `gorm:"primary_key" json:"id"`
	CreatedAt  time.Time      `gorm:"not null" json:"createdAt"`
	UpdatedAt  time.Time      `gorm:"not null" json:"updatedAt"`
	DeletedAt  gorm.DeletedAt `gorm:"index" json:"deletedAt"`
	FeteRoomID uuid.UUID      `gorm:"not null" json:"feteRoomId"`
	FeteRoom   FeteRoom       `gorm:"foreignKey:FeteRoomID"`
	TeamID     uuid.UUID      `gorm:"not null" json:"teamId"`
	Team       FeteTeam       `gorm:"foreignKey:TeamID"`
}

File: js/packages/types/src/feteRoomTeam.ts
import { FeteRoom } from "./feteRoom";
import { FeteTeam } from "./feteTeam";

export interface FeteRoomTeam {
  id: string;
  createdAt: Date;
  updatedAt: Date;
  deletedAt: Date | null;
  feteRoomId: string;
  feteRoom?: FeteRoom;
  teamId: string;
  team?: FeteTeam;
}

